pkg/templates/machinecontroller/helper.go

```diff
@@ -75,3 +79,80 @@ func WaitReady(ctx *util.Context) error {
 	}
 	return nil
 }
+
+// DeleteAllMachines deletes all MachineDeployment, MachineSet and Machine objects.
+func DeleteAllMachines(ctx *util.Context) error {
+	if !ctx.Cluster.MachineController.Deploy {
+		ctx.Logger.Info("Skipping deleting worker machines because machine-controller is disabled in configuration.")
+		return nil
+	}
+	if ctx.DynamicClient == nil {
+		return errors.New("kubernetes client not initialized")
+	}
+
+	bgCtx := context.Background()
+
+	// Annotate nodes with kubermatic.io/skip-eviction=true
+	nodes := &corev1.NodeList{}
+	if err := ctx.DynamicClient.List(bgCtx, &dynclient.ListOptions{}, nodes); err != nil {
+		return errors.Wrap(err, "unable to list nodes")
+	}
+	for i := range nodes.Items {
+		if nodes.Items[i].Annotations == nil {
+			nodes.Items[i].Annotations = map[string]string{}
+		}
+		nodes.Items[i].Annotations["kubermatic.io/skip-eviction"] = "true"
+		if err := ctx.DynamicClient.Update(bgCtx, &nodes.Items[i]); err != nil {
```

Comment:
This is part should contain retry logic, as it's likely that you'll hit a conflict. There is `k8s.io/client-go/util/retry` helper package for retries.

Reply:
Check now. :)